reseq_ros2/reseq_ros2/agevar.py

```diff
@@ -49,8 +47,8 @@ def __init__(self):
 
             # create publisher for the motor topics
             p = self.create_publisher(
-                Motors,
-                f"reseq/module{info['address']}/motor/setpoint",
+                Twist,
+                f"diff_cont_{info['address']}/cmd_vel_unstamped",
```

Comment:
What is listening on the `diff_cont_X` topic? I guess some gazebo package?
However, don't we still need to publish to `reseq/moduleX/motor` to communicate with the robot?

Reply:
As mentioned by @Tizio0o0o0o, a diff_drive_controller is listening to that node. For the physical robot we should therefore also add N diff_drive_node to read from that topic, reconfigured to publish inour motor topics. I'll probably look into that this morning

Reply:
I have to say that I'm not quite sure about this solution anymore, In my opinion a good idea might be to publish a twist command for each module. In this way we are making the system more general (because the module might have their own controller that uses differential or skid steering, differential for now)

The point is that in some place have to exist something that goes from twist to what the robot use (like our current vel_motor function).
- If we are going to use only diff controller the settings are going to be same for every module so...why don't use just the vel_motor function?
- If we are going to use the skid steering controller (not in the near future, but maybe one day...) the settings will be different for each module. In this case a good idea might be to send the twist command to each module, leaving him the role to compute the velocity of the wheels, based on their custom code in low level.

If this idea is the one we choose, we may add a substitute to vel_motor function for gazebo in a different way: another node that subscribe to the topic where twist message are published and publish 4 diff_controller commands in the correct topic ([where gazebo listen to](https://github.com/Team-Isaac-Polito/reseq_ros2/blob/gazebo/reseq_gazebo/config/my_controllers.yaml#L7-L14))

Reply:
You can still create 4 diff_drive and pass the same parameters, and you have the advantage of using a standard ROS node instead of implementing our own. Regarding skid steering and diff drive, consider that in ROS the diff_drive_controller also implements [skid steering](http://wiki.ros.org/diff_drive_controller#Skid_steer) but as we discussed in private it only applies for more than 2 wheels per side.